src/core/or/connection_edge.c

```diff
@@ -2020,16 +2109,15 @@ connection_ap_handshake_rewrite_and_attach(entry_connection_t *conn,
   const int automap = rr.automap;
   const addressmap_entry_source_t exit_source = rr.exit_source;
 
-  /* Now, we parse the address to see if it's an .onion or .exit or
-   * other special address.
-   */
-  const hostname_type_t addresstype = parse_extended_hostname(socks->address);
-
   /* Now see whether the hostname is bogus.  This could happen because of an
    * onion hostname whose format we don't recognize. */
-  if (addresstype == BAD_HOSTNAME) {
+  hostname_type_t addresstype;
+  if (!parse_extended_hostname(socks->address, &addresstype)) {
     control_event_client_status(LOG_WARN, "SOCKS_BAD_HOSTNAME HOSTNAME=%s",
                                 escaped(socks->address));
+    if (addresstype == ONION_V3_HOSTNAME) {
+      conn->socks_request->socks_extended_error_code = SOCKS5_HS_IS_INVALID;
```

Comment:
Hmm, seems to me this is the feature from #30022 but we actually return that the descriptor is invalid. We should make a new error code for when the address is invalid, and this way we also satisfy #30022.

Reply:
Yes! I need to extend prop304 for that and then add that code. I recommend we do that part of #30022 but for sure before we release.

Reply:
ACK. How can we make sure that we don't release in the meanwhile? Because if we release, we will have to deal with weird "if tor_version == XXX then {} else {}" things.

Reply:
043 is not in feature freeze nor close to a stable?